api/src/main/java/com/netflix/iceberg/transforms/Bucket.java

```diff
@@ -114,6 +114,9 @@ public String toString() {
             predicate.op(), name, apply(predicate.literal().value()));
 //      case IN:
 //        return Expressions.predicate();
+      case STARTS_WITH:
+        return Expressions.predicate(
+            predicate.op(), name, apply(predicate.literal().value()));
```

Comment:
This isn't correct. It assumes that `bucket(x) == bucket(y)` if `x.startsWith(y)`.

Reply:
you are totally right, sorry about missing this.
I have updated it.